### PR TITLE
Register sniper upon command interaction instead of on join

### LINKED ISSUE #60: Add a configuration option to disable sniping by deafult
**Is your feature request related to a problem? Please describe.**
Players on my server and I are frustrated when we log in and use the arrow by accident and make small holes everywhere in the map. Most of the time you don't even notice you are deleting blocks (when playing as an archer) that you are removing blocks very far away. These holes have to be cleaned up afterward and other players report them as an issue/bug to our staff team causing us huge amounts of work sometimes.

**Describe the solution you'd like**
A configuration option to disable sniping by default would be a perfect solution.

**Describe alternatives you've considered**
Logging in and typing `/vs disable` every time (current situation) or writing a script that does it for everyone.

### PR DESCRIPTION
**Pull request by StealWonders**
## Overview
**Fixes #60**

## Description
~~I've gone ahead and added a configuration option to disable sniping on login.~~

See comment section: To prevent random sniping when you're not using VS I've gone ahead and changed the point where the sniper is registered from on join to on command.

## Checklist
- [X] I included all information required in the sections above
- [X] I tested my changes and approved their functionality
- [X] I ensured my changes do not break other parts of the code
- [X] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/FastAsyncVoxelSniper/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Review by NotMyFault — CHANGES_REQUESTED** [review]
Instead of having a configuration option, I'd rather default to nothing until the player actually tells Favs what it should do.

**Comment by StealWonders** [comment]
> 
> 
> Instead of having a configuration option, I'd rather default to nothing until the player actually tells Favs what it should do.

Just to clarify: you do not want the default brush (sniping of 1x1 voxel) to be selected but it to rather be "unselected" meaning using the sniping tools yields no effect? Shall active brushes return to an "unselected" state on logout or how long shall they be persisted?

**Comment by NotMyFault** [comment]
> > Instead of having a configuration option, I'd rather default to nothing until the player actually tells Favs what it should do.
> 
> Just to clarify: you do not want the default brush (sniping of 1x1 voxel) to be selected but it to rather be "unselected" meaning using the sniping tools yields no effect? Shall active brushes return to an "unselected" state on logout or how long shall they be persisted?

The idea was to not do anything until the player uses commands, e.g messing with the Fawe wand doesn't do anything with your selection unless you tell it to do something.
Afaik, equipped tools are stored until you unequip them, so they persist thru login sessions until the next reboot.

**Comment by StealWonders** [comment]
> 
> 
> > > Instead of having a configuration option, I'd rather default to nothing until the player actually tells Favs what it should do.
> > 
> > 
> > Just to clarify: you do not want the default brush (sniping of 1x1 voxel) to be selected but it to rather be "unselected" meaning using the sniping tools yields no effect? Shall active brushes return to an "unselected" state on logout or how long shall they be persisted?
> 
> The idea was to not do anything until the player uses commands, e.g messing with the Fawe wand doesn't do anything with your selection unless you tell it to do something.
> Afaik, equipped tools are stored until you unequip them, so they persist thru login sessions until the next reboot.

I still don't understand what you mean by your comment. Brushes in FAWE have to be assigned to tools using `//brush` but in FAVS there is one constantly applied to both gunpowder and the arrow... Comparing the gunpowder and arrow to the WE wand is kind of weird as with the wand you can select stuff and the tools (in VS) apply things directly to the terrain...

**Comment by NotMyFault** [comment]
> I still don't understand what you mean by your comment. Brushes in FAWE have to be assigned to tools using //brush but in FAVS there is one constantly applied to both gunpowder and the arrow

That's exactly what I said & what shouldn't be done. Best to do is simply not assign the Sniper as default.

**Comment by StealWonders** [comment]
Alright, check the latest commit. Snipers will only be created/registered once a player actually interacts with one of the commands. Is this how you'd imagine it? Tested locally and it prevents me from accidentally sniping when I don't want to.
Further question: shall a configuration option be added to delete/remove your sniper session upon logout, as some servers don't restart for a long time.

**Comment by NotMyFault** [comment]
> Alright, check the latest commit. Snipers will only be created/registered once a player actually interacts with one of the commands. Is this how you'd imagine it? Tested locally and it prevents me from accidentally sniping when I don't want to.

Yeah, that's what I meant. No need to do something until told to.

> Further question: shall a configuration option be added to delete/remove your sniper session upon logout, as some servers don't restart for a long time.

That may come in handy for some servers, or those who generally don't want to persist a session over several logins/logouts.

**Comment by NotMyFault** [comment]
Amendment: Such a feature would do better in an additional pr rather in this one 👀

File: src/main/java/com/thevoxelbox/voxelsniper/sniper/SniperRegistry.java [hunk 1]
```diff
@@ -17,6 +17,16 @@ public void register(Sniper sniper) {
         this.snipers.put(uuid, sniper);
     }
 
+    public Sniper registerAndGetSniper(Player player) {
```
Comment (NotMyFault) [inline src/main/java/com/thevoxelbox/voxelsniper/sniper/SniperRegistry.java:27]: Javadocs are missing.

**Review by NotMyFault — CHANGES_REQUESTED** [review]
Otherwise, LGTM.

**Review by NotMyFault — APPROVED** [review]

**Review by Aurelien30000 — APPROVED** [review]
Just a note, be well aware that message on login won’t work until player « equip » a brush.

**Comment by NotMyFault** [comment]
The message on login reflects what you currently have equipped, so if you didn't equip something it wont display something.

**Pull request merged**